legal-api/src/legal_api/services/authz.py

```diff
@@ -95,96 +107,441 @@ def has_roles(jwt: JwtManager, roles: List[str]) -> bool:
+            'registrarsOrder': {
+                'legalTypes':  ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC']
+            },
+            'registration': {
+                'legalTypes': ['SP', 'GP'],
+                'businessExists': False  # only show filing when providing allowable filings not specific to a business
+            },
+            'specialResolution': {
+                'legalTypes': ['CP']
+            },
+            'transition': {
+                'legalTypes': ['BC', 'BEN', 'CC', 'ULC']
+            },
             'restoration': {
-                'limitedRestorationExtension': ['BC', 'BEN', 'CC', 'ULC', 'LLC'],
-                'limitedRestorationToFull': ['BC', 'BEN', 'CC', 'ULC', 'LLC']
+                'limitedRestorationExtension': {
+                    'legalTypes': ['BC', 'BEN', 'CC', 'ULC', 'LLC'],
+                    'blockerChecks': {
+                        'validStateFilings': ['restoration.limitedRestoration',
+                                              'restoration.limitedRestorationExtension'],
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                },
+                'limitedRestorationToFull': {
+                    'legalTypes': ['BC', 'BEN', 'CC', 'ULC', 'LLC'],
+                    'blockerChecks': {
+                        'validStateFilings': ['restoration.limitedRestoration',
+                                              'restoration.limitedRestorationExtension'],
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                }
             }
         },
         Business.State.HISTORICAL: {
-            'courtOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'putBackOn': ['SP', 'GP', 'BEN', 'CP', 'BC', 'CC', 'ULC'],
-            'registrarsNotation': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'registrarsOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
+            'courtOrder': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
+            },
+            'putBackOn': {
+                'legalTypes': ['SP', 'GP', 'BEN', 'CP', 'BC', 'CC', 'ULC'],
+                'blockerChecks': {
+                    # FUTURE add outstanding filings to invalidStateFilings when implemented
+                    'invalidStateFilings': []
+                }
+            },
+            'registrarsNotation': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC']
+            },
+            'registrarsOrder': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC']
+            },
             'restoration': {
-                'fullRestoration': ['CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-                'limitedRestoration': ['BC', 'BEN', 'CC', 'ULC', 'LLC']
+                'fullRestoration': {
+                    'legalTypes': ['CP', 'BC', 'BEN', 'CC', 'ULC'],
```

Comment:
It's nice that the BE supports restoration of a CP, but the FE doesn't handle that yet 😝